NameComparisonToolkit/Name.cs

```diff
@@ -83,36 +82,52 @@ public Name(string firstName, string middleName, string lastName, string suffix)
 	/// <param name="suffix">A string representing the name suffix.</param>
 	public Name(IEnumerable<string> firstName, IEnumerable<string> middleName, IEnumerable<string> lastName, string suffix)
 	{
-		FirstName = firstName ?? Enumerable.Empty<string>();
-		MiddleName = middleName ?? Enumerable.Empty<string>();
-		LastName = lastName ?? Enumerable.Empty<string>();
+		FirstName = firstName;
```

Comment:
I think we want these nullchecks since people could possible new up a name object with null values.

Reply:
no we don't as discussed via slack and I updated the string, string constructor with a full check